deploy/firesim

```diff
@@ -270,11 +291,11 @@ def main(args):
                          args.terminatesomef12, args.terminatesomem416,
                          args.forceterminate)
 
-    if args.task in ['launchrunfarm', 'infrasetup', 'boot', 'kill', 'runworkload', 'runcheck']:
+    if args.task in RUN_TASKS:
         # construct simulation config
         simconf = RuntimeConfig(args)
         # run task
-        globals()[args.task](simconf)
+        RUN_TASKS[args.task](simconf)
```

Comment:
Thinking about @abejgonzalez's question about why I have BUILD_TASKS and RUN_TASKS in two separate datastructures, it would probably be nicer if it was a dict of dicts that also stored the Config class that the task wanted, we could just do this:
```
        f = TASKS[args.task]['function']
        conf = TASKS[args.task]['config_class'](args)
        f(conf)
```
And we could tweak managerinit and terminaterunfarm slightly and not need to have special casing  either.  I like it.

Reply:
@abejgonzalez , I've pushed the rework I did.  You should see that there is only a unified TASKS datastructure now and that register_task is used as a decorator.  I also switched the tests over to using the pytest-mock `mocker` fixture and I think it makes the mock creation look cleaner than using the `@patch` decorator like I was doing before.